File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    heapq.heapify(scoville) # 리스트를 최소힙으로 변환
    count = 0
    
    while len(scoville) >= 2 and scoville[0] < K:
        first = heapq.heappop(scoville)
        second = heapq.heappop(scoville)
        new_scoville = first + second * 2
        heapq.heappush(scoville, new_scoville)
        count += 1
    
    if scoville[0] < K:
        return -1
    
    return count

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int s: scoville) {
            pq.add(s);
        }
        
        int min = pq.peek();
        while (min < K) {
            if (pq.size() >= 2) {
                int first = pq.poll();
                int second = pq.poll();
                int newScoville = first + second * 2;
                pq.add(newScoville);
                min = pq.peek();
                answer ++;
            }
            else return -1;
        }
        return answer;
    }
}